src/BuildScriptGenerator/SdkStorageVersionProviderBase.cs

```diff
@@ -46,17 +47,46 @@ public class SdkStorageVersionProviderBase
             var xdoc = XDocument.Parse(blobList);
             var supportedVersions = new List<string>();
 
-            foreach (var metadataElement in xdoc.XPathSelectElements($"//Blobs/Blob/Metadata"))
+            foreach (var blobElement in xdoc.XPathSelectElements($"//Blobs/Blob"))
             {
-                var childElements = metadataElement.Elements();
-                var versionElement = childElements.Where(e => string.Equals(
-                        versionMetadataElementName,
-                        e.Name.LocalName,
-                        StringComparison.OrdinalIgnoreCase))
-                    .FirstOrDefault();
-                if (versionElement != null)
+                var childElements = blobElement.Elements();
+                if (this.commonOptions.DebianFlavor == OsTypes.DebianStretch)
                 {
-                    supportedVersions.Add(versionElement.Value);
+                    var versionElement = childElements
+                        .Where(e => string.Equals("Metadata", e.Name.LocalName, StringComparison.OrdinalIgnoreCase))
+                        .FirstOrDefault()?.Elements()
+                        .Where(e => string.Equals(SdkStorageConstants.LegacySdkVersionMetadataName, e.Name.LocalName, StringComparison.OrdinalIgnoreCase))
+                        .FirstOrDefault();
+
+                    if (versionElement != null)
+                    {
+                        supportedVersions.Add(versionElement.Value);
+                    }
+
+                    continue;
+                }
+
+                // TODO: PR2 Remove this logic, as we will be able to depend on bullseye images existing in the sdk storage
+                var patternText = this.commonOptions.DebianFlavor == OsTypes.DebianBullseye
+                    ? $"{platformName}-buster-(?<version>.*?).tar.gz"
```

Comment:
Same comment here: is `bullseye` being treated as `buster` for the time being?

(also same comment about using the `OsTypes` constants)

Reply:
Yes thats right, for the time being we want bullseye images to pull buster sdks, until this PR releases bullseye sdks to our storage account.